d2common/d2loader/mpq/asset.go

```diff
@@ -35,14 +42,58 @@ func (a *Asset) Path() string {
 
 // Read will read asset data into the given buffer
 func (a *Asset) Read(buf []byte) (n int, err error) {
-	return a.stream.Read(buf)
+	totalRead, err := a.stream.Read(buf)
+	if totalRead == 0 {
+		return 0, io.EOF
+	}
+
+	return totalRead, nil
 }
 
 // Seek will seek the read position for the next read operation
 func (a *Asset) Seek(offset int64, whence int) (n int64, err error) {
 	return a.stream.Seek(offset, whence)
 }
 
+// Seek will seek the read position for the next read operation
+func (a *Asset) Close() (err error) {
+	_, err = a.Seek(0, 0)
```

Comment:
I think this should be `a.stream.Close()`

Reply:
So this is a strange issue i was having. Usually, calling close will set a stream to nil, and if i call the mpq stream `Close` method it will do the same. But, because i am caching these mpq file entries for retrieval later, i don't close the stream, i just seek to 0 and keep it open. The next time this entry is retrieved from the cache, it will be ready to read from. I think the only time it will be closed is after it is ejected from the cache and gc gets to it.

Reply:
The asset needs to implement `io.Reader`, `io.Writer`, and `io.Seeker`, an unexported `reset` method would just be a wrapper for `Seek(0, 0)`. I've opted to just add a comment here that explains what the magic is.